Add path checks before scheduling and before deletion (#149)
ExpediaGroup/beekeeper: CHANGELOG.md

The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).

## [3.5.0] - TBD
### Changed

> - Added additional checks for the number of levels in table and partition paths so that invalid paths don't get scheduled for deletion.

Reviewer: ```suggestion
- Added additional checks for the number of levels in table and partition paths so that invalid paths are not scheduled for deletion.
```
